neuronpp/core/cells/spine_cell.py

```diff
@@ -78,3 +120,369 @@ def _connect_necks_rand_uniform(neck: Sec, sections):
+           if int draw spine position from the uniform distribution
+        """
+        if not isinstance(section, Sec):
+            section = Sec(section)
+        if isinstance(u_random, int):
+            target_locations = np.random.uniform(0., 1., n_spines).tolist()
+        else:
+            target_locations = np.linspace(0., .99, n_spines).tolist()
+
+        self._add_spines_to_section(section, spine_name,
+                                    target_locations, head_diam,
+                                    head_len, neck_diam, neck_len,
+                                    E_pas, g_pas, ra, cm,
+                                    add_pas=add_pas)
+        return target_locations
+
+    def _add_spines_to_section(self, section, spine_name,
+                               target_location, head_diam,
+                               head_len, neck_diam, neck_len,
+                               E_pas, g_pas, ra, cm, add_pas=True):
+        name = section.name
+        if not isinstance(target_location, list):
+            target_location = [target_location]
+        for i, location in enumerate(target_location):
+            head = self.add_sec(name="%s_%s_head[%d]" % (name, spine_name, i),
+                                diam=head_diam, l=head_len, nseg=2,
+                                E_rest=E_pas, ra=ra, cm=cm,
+                                g_pas=g_pas, add_pas=add_pas)
+            neck = self.add_sec(name="%s_%s_neck[%d]" % (name, spine_name, i),
+                                diam=neck_diam, l=neck_len, nseg=1,
+                                E_rest=E_pas, ra=ra, cm=cm,
+                                g_pas=g_pas, add_pas=add_pas)
+            self.heads.append(head)
+            self.necks.append(neck)
+            spine = Spine(head, neck, self, "%s_spine_%s[%d]" % (name,
+                                                                 spine_name,
+                                                                 i))
+            self.spines.append(spine)
+            self.connect_secs(source=neck, target=section, source_loc=location,
+                              target_loc=0.0)
+
+    def find_sections_with_mech(self, mech_name):
+        mech_dend_loc = {}
+        all_spines = self.spines
+
+        for spine in all_spines:
+            spine_name = spine.name.split(".")[-1]
+            spine_mechs = set(list(spine.neck.hoc.psection()["density_mechs"])+\
+                               list(spine.head.hoc.psection()["density_mechs"]))
+
+            if mech_name is None or mech_name in spine_mechs:
+                parent_mechs = spine.parent.hoc.psection()["density_mechs"]
+                if mech_name is None or mech_name in parent_mechs:
+                    if spine.parent not in mech_dend_loc:
+                        mech_dend_loc[spine.parent] = []
+                    mech_dend_loc[spine.parent].append(spine)
+        return mech_dend_loc
+
+    def _get_spine_factor(self, spines, mech_name, gbar=None):
```

Comment:
although it is a private method, could you ad at least short docstring? Because it is not clear for me from the name what the method is doing - what is the spine factor?